config/config-api/src/com/thoughtworks/go/config/AgentConfig.java

```diff
@@ -165,6 +170,10 @@ public boolean isNull() {
         return false;
     }
 
+    public AgentConfig deepClone() {
+        return new Cloner().deepClone(this);
```

Comment:
We get the pending agent from AgentInstances object which is a live instance and we do not want to modify it. That is why we deep clone the object and perform modifications on top of that.